src/server/views/dashboard/macros.njk

```diff
@@ -50,85 +51,88 @@
 {% endmacro %}
 
 {% macro listItemEditActionBar(list, listItem, dashboardRoutes, titleProperty, isPinned, actionButtons, changeMessage) %}
+  {% set actionToLabelText = {
+      publish: "Publish",
+      remove: "Remove",
+      requestChanges: "Request changes",
+      unpin: "Unpin",
+      unpublish: "Unpublish",
+      updateLive: "Update live version",
+      updateNew: "Publish",
+      archive: "Archive"
+    }
+  %}
 
+  {% set changeMessageHtml %}
+    {{ govukTextarea ({
+      id: "message",
+      name: "message",
+      label: {
+        text: "Change message",
+        classes: "govuk-visually-hidden"
+      },
+      hint: {
+        text: "Provide simple and concise instructions to the service provider on what they need to change."
+      }
+    }) }}
+  {% endset %}
+
+  {% set archiveHtml %}
+    {{ govukTextarea ({
+      id: "reason",
+      name: "reason",
+      label: {
+        text: "Reason",
+        classes: "govuk-visually-hidden"
+      },
+      hint: {
+        text: "Reason for archiving"
+      }
+    }) }}
+  {% endset %}
+
+  {% set items = [] %}
+
+  {% for action in actionButtons %}
+    {% if action in ["requestChanges", "archive"] %}
+      {% set isRequestChanges = action == "requestChanges" %}
+      {% set item = {
+        text: actionToLabelText[action],
+        value: action,
+        id: action,
+        conditional: {
+          html: changeMessageHtml if isRequestChanges else archiveHtml
+        }
+      } %}
+    {% else %}
+      {% set item = {
+        text: actionToLabelText[action],
+        value: action,
+        id: action
+      } %}
+    {% endif %}
+
+
+    {% set items = (items.push(item), items)  %}
+
+  {% endfor %}
   <div class="govuk-!-margin-top-8">
-  <div class="govuk-form-group">
```

Comment:
I have changed this to use govuk njk components. 

It makes it a lot easier/managable/readable when dealing with a lot of (related) inputs. The components are fairly flexible. You can control idPrefix, name, attributes, conditional fields, fieldset, legend, etc. 

Take a look at the njk options tab 
https://design-system.service.gov.uk/components/radios

![image](https://user-images.githubusercontent.com/22080510/219686076-34c8ddc8-396c-45c3-96f3-8cf0f75d5038.png)